LV_analysis/1_get_multiplier_LV_coverage_Pa.py

```diff
@@ -0,0 +1,335 @@
+    ).merge(
+        pd.DataFrame(
+            dict_highweight_coverage[gene_label], columns=["highweight LV coverage"]
+        ),
+        left_index=True,
+        right_index=True,
+    )
+    merged_df["gene type"] = gene_label
+    all_coverage.append(merged_df)
+
+all_coverage_df = pd.concat(all_coverage)
+# -
+
+all_coverage_df = lv.assemble_coverage_df(
+    processed_dict_genes, dict_nonzero_coverage, dict_highweight_coverage
+)
+all_coverage_df.head()
+
+# Plot coverage distribution given list of generic coverage, specific coverage
+nonzero_fig = sns.boxplot(
+    data=all_coverage_df,
+    x="gene type",
+    y="nonzero LV coverage",
+    notch=True,
+    palette=["#2c7fb8", "lightgrey"],
+)
+nonzero_fig.set_xlabel(None)
+nonzero_fig.set_xticklabels(
+    ["generic genes", "other genes"], fontsize=14, fontname="Verdana"
+)
+nonzero_fig.set_ylabel(
+    textwrap.fill("Number of LVs", width=30), fontsize=14, fontname="Verdana"
+)
+nonzero_fig.tick_params(labelsize=14)
+nonzero_fig.set_title(
+    "Number of LVs genes are present in", fontsize=16, fontname="Verdana"
+)
+
+# Plot coverage distribution given list of generic coverage, specific coverage
+highweight_fig = sns.boxplot(
+    data=all_coverage_df,
+    x="gene type",
+    y="highweight LV coverage",
+    notch=True,
+    palette=["#2c7fb8", "lightgrey"],
+)
+highweight_fig.set_xlabel(None)
+highweight_fig.set_xticklabels(
+    ["generic genes", "other genes"], fontsize=14, fontname="Verdana"
+)
+highweight_fig.set_ylabel(
+    textwrap.fill("Number of LVs", width=30), fontsize=14, fontname="Verdana"
+)
+highweight_fig.tick_params(labelsize=14)
+highweight_fig.set_title(
+    "Number of LVs genes contribute highly to", fontsize=16, fontname="Verdana"
+)
+
+# ## Calculate statistics
+# * Is the reduction in generic coverage significant?
+# * Is the difference between generic versus other genes signficant?
+
+# +
+# Test: mean number of LVs generic genes present in vs mean number of LVs that generic gene is high weight in
+# (compare two blue boxes between plots)
+generic_nonzero = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "nonzero LV coverage"
+].values
+generic_highweight = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "highweight LV coverage"
+].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(generic_nonzero, generic_highweight)
+print(pvalue)
+
+# +
+# Test: mean number of LVs generic genes present in vs mean number of LVs other genes high weight in
+# (compare blue and grey boxes in high weight plot)
+other_highweight = all_coverage_df[all_coverage_df["gene type"] == "other"][
+    "highweight LV coverage"
+].values
+generic_highweight = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "highweight LV coverage"
+].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(other_highweight, generic_highweight)
+print(pvalue)
+
+# +
+# Check that coverage of other and generic genes across all LVs is NOT signficantly different
```

Comment:
So these _are_ significantly different here, right, since the p-value is small? Meaning generic genes on average contribute to fewer LVs than specific genes do?

Reply:
Yes. you're interpretation is correct here. More below